src/pyOpenMS/setup_new_approach.py

```
@@ -0,0 +1,310 @@
+                cmake_options.append('-D{0}={1}'.format(
+                    varname, 'on' if value else 'off'))
+
+            def append_cmake_component(flag, varname):
+                # only pass this to cmake if the user pass the --with-component
+                # flag to setup.py build_ext
+                if flag is not None:
+                    flag_name = (
+                        "--with-"
+                        + varname.removeprefix("OPENMS_").lower().replace("_", "-"))
+                    warnings.warn(
+                        MSG_DEPR_SETUP_BUILD_FLAGS.format(flag_name),
```

Ruff (0.8.2)

Undefined name `MSG_DEPR_SETUP_BUILD_FLAGS`

(F821)